plugin/external/README.md

```diff
@@ -0,0 +1,78 @@
+# external
+
+## Name
+
+*external* - resolve ingress and load balance IPs from outside kubernetes clusters.
+
+## Description
+
+This plugin allows an additional zone to resolve the external IP address(es) of a Kubernetes
+service. This plugin is only useful if the *kubernetes* plugin is also loaded.
+
+The plugin uses an external zone to resolve in-cluster IP addresses. It only handles queries for A,
+AAAA and SRV records, all others result in NODATA responses. To make it a proper DNS zone it handles
+SOA and NS queries for the apex of the zone.
+
+By default the apex of the zone will look like (assuming the zone used is `example.org`):
+
+~~~ dns
+example.org.	5 IN	SOA ns1.dns.example.org. hostmaster.example.org. (
+				12345      ; serial
+				14400      ; refresh (4 hours)
+				3600       ; retry (1 hour)
+				604800     ; expire (1 week)
+				5          ; minimum (4 hours)
+				)
+example.org		5 IN	NS ns1.dns.example.org.
+
+ns1.dns.example.org.  5 IN  A    ....
+ns1.dns.example.org.  5 IN  AAAA ....
+~~~
+
+Note we use the `dns` subdomain to place the records the DNS needs (see the `apex` directive). Also
+note the SOA's serial number is static. The IP addresses of the nameserver records are those of the
+CoreDNS service.
+
+The *external* plugin handles the subdomain `dns` and the apex of the zone by itself, all other
+queries are resolved to addresses in the cluster.
+
+## Syntax
+
+~~~
+external [ZONE...]
```

Comment:
Can we think of a different name? I mean, "external" is very general and this is really k8s specific.

Reply:
done